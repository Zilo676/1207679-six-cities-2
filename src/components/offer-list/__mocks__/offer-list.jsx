import React from "react";

const OfferList = () => {
  return <></>;
};

export {OfferList};
export default OfferList;
