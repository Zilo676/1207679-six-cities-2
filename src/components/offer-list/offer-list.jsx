import React from "react";
import PropTypes from 'prop-types';

import {OfferCard} from '../offer-card/offer-card.jsx';

const OfferList = (props) => {
  const {offers, onActiveItem} = props;

  return (
    <div className="cities__places-list places__list tabs__content">
      {offers.map((offer) => {
        return (
          <OfferCard
            key={offer.id}
            offer={offer}
            onOfferCardHover={() => onActiveItem(offer)}
          />
        );
      })
      }
    </div>
  );

};

OfferList.propTypes = {
  offers: PropTypes.arrayOf(PropTypes.shape(
      {
        id: PropTypes.number.isRequired,
        city: PropTypes.shape({
          name: PropTypes.string.isRequired,
          location: PropTypes.shape({
            latitude: PropTypes.number.isRequired,
            longitude: PropTypes.number.isRequired,
            zoom: PropTypes.number.isRequired,
          })
        }),
        [`preview_image`]: PropTypes.string.isRequired,
        images: PropTypes.arrayOf(PropTypes.string),
        title: PropTypes.string.isRequired,
        [`is_favorite`]: PropTypes.bool.isRequired,
        [`is_premium`]: PropTypes.bool.isRequired,
        rating: PropTypes.number.isRequired,
        type: PropTypes.string.isRequired,
        bedrooms: PropTypes.number.isRequired,
        [`max_adults`]: PropTypes.number.isRequired,
        price: PropTypes.number.isRequired,
        goods: PropTypes.arrayOf(PropTypes.string.isRequired),
        host: PropTypes.shape({
          id: PropTypes.number.isRequired,
          [`is_pro`]: PropTypes.bool.isRequired,
          name: PropTypes.string.isRequired,
          [`avatar_url`]: PropTypes.string.isRequired,
        }),
        description: PropTypes.string.isRequired,
        location: PropTypes.shape({
          latitude: PropTypes.number.isRequired,
          longitude: PropTypes.number.isRequired,
          zoom: PropTypes.number.isRequired,
        })
      }
  ).isRequired
  ).isRequired,
  onActiveItem: PropTypes.func.isRequired,
};

export {OfferList};
