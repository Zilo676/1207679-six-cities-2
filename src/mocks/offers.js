const offers = [
  {
    id: `799663af-e554-49c5-b73e-d61b36b503f0`,
    type: `Apartament`,
    price: 120,
    description: `Beautiful & luxurious apartment at great location`,
    coordinates: [52.3909553943508, 4.85309666406198],
    raiting: 93,
    city: `Amsterdam`,
  },
  {
    id: `4a6cf5ff-150e-40f2-baf8-820461958a61`,
    type: `Private room`,
    price: 80,
    description: `Wood and stone place`,
    coordinates: [52.369553943508, 4.85309666406198],
    raiting: 80,
    city: `Amsterdam`,
  },
  {
    id: `0dcee833-d3f2-4f12-9559-a2775ca7f733`,
    type: `Apartament`,
    price: 132,
    description: `Canal View Prinsengracht`,
    coordinates: [52.3909553943508, 4.929309666406198],
    raiting: 80,
    city: `Amsterdam`,
  },
  {
    id: `f5035346-5f36-480f-9401-b8f5dc8186c6`,
    type: `Apartament`,
    price: 180,
    description: `Nice, cozy, warm big bed apartment`,
    coordinates: [52.3809553943508, 4.939309666406198],
    raiting: 100,
    city: `Amsterdam`,
  },
  {
    id: `76119596-1f8c-4df9-bb83-ce109ef8f3af`,
    type: `Apartament`,
    price: 180,
    description: `Nice, cozy, warm big bed apartment`,
    coordinates: [48.864716, 2.349014],
    raiting: 100,
    city: `Paris`,
  },
  {
    id: `76119596-1f8c-4df9-bb83-ce109ef8f3af`,
    type: `Apartament`,
    price: 180,
    description: `Nice, cozy, warm big bed apartment`,
    coordinates: [50.935173, 6.953101],
    raiting: 100,
    city: `Colonge`,
  },
  {
    id: `76119596-1f8c-4df9-bb83-ce109ef8f3af`,
    type: `Apartament`,
    price: 180,
    description: `Nice, cozy, warm big bed apartment`,
    coordinates: [50.85045, 4.34878],
    raiting: 100,
    city: `Brussels`,
  },
  {
    id: `76119596-1f8c-4df9-bb83-ce109ef8f3af`,
    type: `Apartament`,
    price: 180,
    description: `Nice, cozy, warm big bed apartment`,
    coordinates: [53.551086, 9.993682],
    raiting: 100,
    city: `Hamburg`,
  },
  {
    id: `76119596-1f8c-4df9-bb83-ce109ef8f3af`,
    type: `Apartament`,
    price: 180,
    description: `Nice, cozy, warm big bed apartment`,
    coordinates: [51.2277411, 6.7734556],
    raiting: 100,
    city: `Dusseldorf`,
  }
];

export {offers};
