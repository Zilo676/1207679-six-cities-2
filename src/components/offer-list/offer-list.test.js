import React from 'react';
import renderer from 'react-test-renderer';

import {OfferList} from './offer-list.jsx';
import {offers} from '../../mocks/offers.js';

it(`Correctly rendered`, () => {
  const tree = renderer.create(<OfferList offers={offers}/>).toJSON();
  expect(tree).toMatchSnapshot();
});
