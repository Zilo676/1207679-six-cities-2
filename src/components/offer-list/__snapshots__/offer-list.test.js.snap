// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Correctly rendered 1`] = `
<div
  className="cities__places-list places__list tabs__content"
>
  offer-card
  offer-card
</div>
`;
